test/integration/features/basic.feature

```diff
@@ -10,6 +10,61 @@ Feature: Basic
      Then Minishift should have state "Running"
       And Minishift should have a valid IP address
 
+  Scenario: User is able to do ssh into Minishift VM
+    Given Minishift has state "Running"
+     When executing "minishift ssh echo hello"
+     Then stderr should be empty
+      And exitcode should equal 0
+      And stdout should contain
+      """
+      hello
+      """
+
+  Scenario: User is able to set custom Docker specific environment variables
+    Given Minishift has state "Running"
+     When executing "minishift ssh cat /var/lib/boot2docker/profile"
+     Then stderr should be empty
+      And exitcode should equal 0
+      And stdout should contain
+      """
+      export "FOO=BAR"
+      export "BAZ=BAT"
+      """
+
+  Scenario Outline: User can set, get, view and unset values in configuration file
+    User is able to setup persistent configuration of Minishift using "config" command
+    and its subcommands.
+    Given Minishift has state "Running"
+     When executing "minishift config set <property> <value>"
+     Then JSON config file contains key "<property>" with value "<value>"
```

Comment:
We need to find a balance here. To generic steps might be awkward to use, but to small scoped steps means one needs to write actual step implementations for each single thing one wants to test.